[UI] Wallet repair option to resync from scratch

Mrs-X opened the pull request:
This option was requested by @s3v3nh4cks , so blame him if you don't like it :-)

When executed, this repair option restarts the wallet with the parameter `-resync` and will delete the folders `blocks`, `chainstate`, `sporks` and `zerocoin` and the wallet starts to synchronize from scratch.
This should help users to reinitialize their local blockchain-data into a consistent state without the need to explain where those folders are and which folders to delete.

It's also available as command line parameter `-resync` if the wallet-UI isn't usable for whatever reasons.

**Please(!) test this on OSX and Windows and report back (both positive and negative results)**

Since resync needs a lot of time and temporary missing funds might scare users I've added an appropriate warning:

![resync](https://user-images.githubusercontent.com/22873440/32022897-0a072f26-b9d8-11e7-8cf8-d47e3e334bf5.png)

Fuzzbawls commented on src/init.cpp:
```diff
@@ -1043,6 +1043,40 @@ bool AppInit2(boost::thread_group& threadGroup)
+            filesystem::path sporksDir = GetDataDir() / "sporks";
+            filesystem::path zerocoinDir = GetDataDir() / "zerocoin";
+            
+            LogPrintf("Deleting blockchain folders blocks, chainstate, sporks and zerocoin\n");
```
Nit: Even though this process should be short lived, possible to add an InitMessage here as well?

`uiInterface.InitMessage(_("Preparing for resync..."));`

Fuzzbawls commented on src/init.cpp:
```diff
@@ -1043,6 +1043,40 @@ bool AppInit2(boost::thread_group& threadGroup)
             }
         }
 
+        if (GetBoolArg("-resync", false)) {
```
This new command-line parameter will need to be documented further up as a `HelpMessageOpt()` or it will fail the documentation check.

Mrs-X commented:
`HelpMessageOpt()` good point, missed this.
I've also added the `uiInterface.InitMessage()` , even when it's only visible for a fraction of a second on a halfway decent computer...

Fixed and squashed.

jonspock commented:
ACK. Tested from the command line. I was under the impression from the text that I would still get a Warning Pop up before proceeding. Also tested with testnet flag.

Mrs-X commented on src/qt/forms/rpcconsole.ui:
```diff
@@ -1077,10 +1077,23 @@
+          </size>
+         </property>
+         <property name="text">
+          <string>Delete lokal Blockchain Folders</string>
```
Fixed and squashed.

Mrs-X commented:
@jonspock I thought about this also, but in my opinion the target group for the popup-warning are exactly those users which don't know or want the command line.
Power users on the other side will complain when they do things automatically via scripts and a popup would need user-interaction.

jonspock commented:
Agree, not needed for command line, more of just my misunderstanding of the text

Mrs-X commented:
Any Windows testers around?

jonspock commented:
Make sure you find that one Windows tester who builds natively on Mingw while you are at it!

PIVX-Project commented:
utACK

Fuzzbawls commented:
ACK 1393e37

UI and CLI both working as intended on Windows
